src/app/public/modules/phone-field/phone-field-input.directive.ts

```diff
@@ -0,0 +1,232 @@
+import {
+  Directive,
+  forwardRef,
+  ElementRef,
+  EventEmitter,
+  Input,
+  HostListener,
+  AfterViewInit,
+  Injector,
+  ChangeDetectorRef,
+  OnInit,
+  Output
+} from '@angular/core';
+
+import {
+  NG_VALUE_ACCESSOR,
+  NG_VALIDATORS,
+  ControlValueAccessor,
+  Validator,
+  AbstractControl,
+  NgControl,
+  FormControl
+} from '@angular/forms';
+
+import {
+  SkyCountryData
+} from './types';
+
+import {
+  SkyPhoneFieldAdapterService
+} from './phone-field-adapter.service';
+
+import {
+  SkyPhoneFieldComponent
+} from './phone-field.component';
+
+// tslint:disable:no-forward-ref no-use-before-declare
+const SKY_PHONE_FIELD_VALUE_ACCESSOR = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => SkyPhoneFieldInputDirective),
+  multi: true
+};
+
+const SKY_PHONE_FIELD_VALIDATOR = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => SkyPhoneFieldInputDirective),
+  multi: true
+};
+// tslint:enable
+
+@Directive({
+  selector: '[skyPhoneFieldInput]',
+  providers: [
+    SKY_PHONE_FIELD_VALUE_ACCESSOR,
+    SKY_PHONE_FIELD_VALIDATOR,
+    SkyPhoneFieldAdapterService
+  ]
+})
+export class SkyPhoneFieldInputDirective implements OnInit, AfterViewInit,
+  ControlValueAccessor, Validator {
+
+  /**
+   * Disabling linting here as per the Angular style guide "Style 05-13" it is acceptable
+   * to alias when when the directive name is also an input property,
+   * and the directive name doesn't describe the property.
+   */
+  // tslint:disable-next-line:no-input-rename
+  @Input('skyPhoneFieldInput')
+  public skyPhoneFieldComponent: SkyPhoneFieldComponent;
+
+  @Input()
+  public set defaultCountry(value: string) {
+    this.skyPhoneFieldComponent.setDefaultCountry(value);
+    this._defaultCountryCode = value;
+  }
+
+  public get defaultCountry() {
+    return this._defaultCountryCode;
+  }
+
+  @Input()
+  public set disabled(value: boolean) {
+    this.skyPhoneFieldComponent.disabled = value;
+    this.adapterService.setElementDisabledState(this.elRef.nativeElement, value);
+    this._disabled = value;
+  }
+
+  public get disabled(): boolean {
+    return this._disabled || false;
+  }
+
+  @Input()
+  public formatModel: boolean = true;
+
+  @Input()
+  public noValidate: boolean = false;
```

Comment:
Is 'noValidate` a standard naming convention? The default scenario of `noValidate = false` creates a confusing double negative. Should it just be `validate` instead? Or is there something better?

Reply:
It is. That property is often used on forms and is built in there. However, you can see where we have a custom `noValidate` on the datepicker as well. I'm wondering if I should rename the property to `skyPhoneFieldNoValidate` though since `noValidate` is technically a HTML key word that doesn't require a `true/false`. What do you think? 

I was attempting to make things clearer by not having the prefix but I'm wondering if that will make it more confusing.

Reply:
Ah, yes I see the `noValidate` attribute now, thanks for sorting me out. hmm.. I really don't know which is better.  I think we're pushing towards prefixing everything with `sky` so maybe that is the better route?

Reply:
Renamed